src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -31,19 +35,30 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
 import org.bson.BsonDocument;
 import org.bson.BsonInt32;
 import org.bson.Document;
 
+@AutoLoggable
 public abstract class MongoConnection implements Connection {
     protected MongoClient mongoClient;
     protected String currentDB;
     protected String url;
     protected String user;
     protected boolean isClosed;
+    protected MongoLogger logger;
+    protected int connectionId;
+    private static AtomicInteger connectionCounter = new AtomicInteger();
+    private AtomicInteger stmtCounter = new AtomicInteger();
```

Comment:
This is because each Connection has its own set of statements so I thought it was making more sense to use a separate counter.
Otherwise, if connection 1 creates 2 statements and then connection 2 creates 1 statement instead of [c-1][s-1], [c-1][s-2], [c-2][s-1] we will have [c-1][s-1], [c-1][s-2], [c-2][s-3] which can be a little harder to track in the application code.